sdk/core/core-client/src/authorizeRequestOnClaimChallenge.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions } from "@azure/core-auth";
+import { AuthorizeRequestOnChallengeOptions } from "@azure/core-rest-pipeline";
+import { createClientLogger } from "@azure/logger";
+import { decodeStringToString } from "./base64";
+
+const logger = createClientLogger("authorizeRequestOnClaimChallenge");
```

Comment:
it feels weird to create a new package logger for this - don't SDKs usually create one with the package name instead? Or is this special because it's basically a mixin function? Perhaps the logger should be passed in instead?

Reply:
I like the idea of accepting a logger to be passed in! I’ll make a PR

Reply:
I have made a PR: https://github.com/Azure/azure-sdk-for-js/pull/18467